tensorflow/lite/micro/kernels/split_v.cc

```diff
@@ -0,0 +1,173 @@
+  int axis = axis_value < 0 ? axis_value + split_dimensions : axis_value;
+
+  TFLITE_DCHECK_LT(axis, split_dimensions);
+  TFLITE_DCHECK_EQ(output_dims->size, split_dimensions);
+
+  int64_t split_size = 0;
+  for (int i = 0; i < outputs_count; i++) {
+      split_size += tflite::micro::GetEvalOutput(context, node, i)->dims->data[axis];
+  }
+
+  TFLITE_DCHECK_EQ(split_size, input_dims->data[axis]);
+  int64_t outer_size = 1;
+  for (int i = 0; i < axis; ++i) {
+    outer_size *= input_dims->data[i];
+  }
+
+  int64_t base_inner_size = 1;
+  for (int i = axis + 1; i < split_dimensions; ++i) {
+    base_inner_size *= input_dims->data[i];
+  }
+  
+  const T* input_ptr = tflite::micro::GetTensorData<T>(input);
+  for (int k = 0; k < outer_size; ++k) {
+    for (int i = 0; i < outputs_count; ++i) {
+      TfLiteEvalTensor* t = tflite::micro::GetEvalOutput(context, node, i);
+      T* output_data = tflite::micro::GetTensorData<T>(t);
+      output_dims = t->dims;
+      const int copy_size = output_dims->data[axis] * base_inner_size;
+      T* output_ptr = output_data + k * copy_size;
+      for (int j = 0; j < copy_size; ++j) output_ptr[j] = input_ptr[j];
+      input_ptr += copy_size;
+    }
+  }
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
+  TF_LITE_ENSURE_EQ(context, NumInputs(node), 3);
+
+  OpContext op_context(context, node);
+
+  //TF_LITE_ENSURE_EQ(context, NumOutputs(node), op_context.params->num_splits);
+
+  auto input_type = op_context.input->type;
+  TF_LITE_ENSURE(context,
+                 input_type == kTfLiteFloat32 || input_type == kTfLiteUInt8 ||
+                     input_type == kTfLiteInt16 || input_type == kTfLiteInt32 ||
+                     input_type == kTfLiteInt64 || input_type == kTfLiteInt8);
+  for (int i = 0; i < NumOutputs(node); ++i) {
+    GetOutput(context, node, i)->type = input_type;
+  }
+
+   return kTfLiteOk;
+}
+
+TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
+  OpContext op_context(context, node);
+ 
+  const TfLiteTensor* axis_tensor = GetInput(context, node, 2);
+  const TfLiteTensor* size_splits_tensor = GetInput(context, node, 1);
+  // if not const return error - unsupported
+  if (!IsConstantTensor(axis_tensor) || !IsConstantTensor(size_splits_tensor)) {
+    TF_LITE_KERNEL_LOG(
+        context, "Only const tensors are supported for size_splits and axis");
+    return kTfLiteError;
+  }
+ 
+
+  switch (op_context.input->type) {
+    case kTfLiteFloat32: {
+      SplitImpl<float>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteUInt8: {
+      SplitImpl<uint8_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt16: {
+      SplitImpl<int16_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt32: {
+      SplitImpl<int32_t>(context, node, &op_context);
+      break;
+    }
+    case kTfLiteInt64: {
+      SplitImpl<int64_t>(context, node, &op_context);
+      break;
+    }
```

Comment:
can you add more context on why types other than float32 and int8 are necessary? in general, for micro we would like to limit support for only what is strictly needed.

Reply:
Since the uint8 quantization is now legacy (which is creating more issues than value). If you have a production model that needs uint8, then its ok to keep. If at all possible, I would request removing it as we are attempting to steer people away from it.

Ok to keep int16 and int32 from my perspective.

Reply:
That's fine I think. uint8 removed :)